Changes for SDDP.jl v1.0

odow opened the pull request.

odow committed:
Fix links to documentation in README.md (#5)

odow committed:
Bump versions in GitHub action YAML files (#3)

odow committed:
Add missing license headers (#2)

odow committed:
Replace GLPK with HiGHS in the tests and add compat fields for all packages (#1)

adow031 reviewed (approved):
These changes look good.

odow commented:
Do you want to review individually? Or should I just merge a bunch into this branch? 

See https://github.com/EPOC-NZ/JADE.jl/pull/14#issuecomment-1349941555

adow031 commented:
> Do you want to review individually? Or should I just merge a bunch into this branch?
> 
> See [#14 (comment)](https://github.com/EPOC-NZ/JADE.jl/pull/14#issuecomment-1349941555)

Sorry. I didn't see your earlier comment. This one should be merged onto the main branch since it fixes the links. The others can go onto develop.

odow committed:
Remove unused LinearAlgebra package (#14)

odow committed:
Remove unused DataFrames package (#15)

odow committed:
Refactor tests into functional form and remove global state (#21)

odow committed:
Remove InSampleMonteCarlo2 (#23)

odow committed:
Audit the code in time.jl (#17)

Doing this was actually quite sensible, because our year-week notion
of time is a bit more complicated to do in Dates.jl.

There were few problems:

 * I think in early versions of Julia you needed to import a function
   from Base to extend it. Now you can just prefix with Base.
 * There were a bunch of functions not actually used or tested

odow committed:
Remove WrapHistorical sampling scheme with _TerminateOnCycle (#20)

odow committed:
Remove use of global variables (#26)

odow committed:
Switch packages to import to force namespacing their methods (#25)

This is good developement practice because it makes it easier to find
which packages are used where in the codebase.

odow committed:
Remove CRC32c (#28)

odow committed:
Remove DelimitedFiles (#29)

odow committed:
Remove JADEForwardPass and _TerminateOnCycle and update to SDDP v0.4.8 (#27)

odow committed:
Rename JADE_DIR to __JADE_DIR__ and add a docstring (#18)

odow committed:
Update to SDDP@1 (#32)

odow committed:
Remove modified method SDDP.write_cuts_to_file (#33)

odow committed:
Tidy format of src/data/checks.jl (#35)

odow committed:
Refactor data parsing to use CSV.jl and add tests (#34)
